index.js

```diff
@@ -306,50 +351,101 @@ class GithubScm extends Scm {
-            // eslint-disable-next-line no-underscore-dangle
-            this._getBranchUrl(scmInfo, config.token)
-        ]).then(([repoInfo, branchUrl]) => ({
-            id: `${scmInfo.host}:${repoInfo.id}:${scmInfo.branch}`,
-            name: repoInfo.full_name,
-            url: branchUrl
-        }));
+                return resolve({
+                    avatar: data.avatar_url,
+                    name: data.name,
+                    username: data.login,
+                    url: data.html_url
+                });
+            });
+        });
     }
 
     /**
-     * Given a SCM webhook payload & its associated headers, aggregate the
-     * necessary data to execute a Screwdriver job with.
-     *
-     * Temporary shiv until 2.0 is implemented
-     * @method parseHook
-     * @param  {Object}  payloadHeaders  The request headers associated with the
-     *                                   webhook payload
-     * @param  {Object}  webhookPayload  The webhook payload received from the
-     *                                   SCM service.
-     * @return {Promise}
+     * Decorate the commit based on the repository
+     * @method _decorateCommit
+     * @param  {Object}        config        Configuration object
+     * @param  {Object}        config.scmUri SCM URI the commit belongs to
+     * @param  {Object}        config.token  Service token to authenticate with Github
+     * @return {[type]}               [description]
      */
-    parseHook(payloadHeaders, webhookPayload) {
-        return new Promise((resolve, reject) => {
-            try {
-                // eslint-disable-next-line no-underscore-dangle
-                const result = this._parseHook(payloadHeaders, webhookPayload);
+    _decorateCommit(config) {
+        return this.lookupScmUri({
+            scmUri: config.scmUri,
+            token: config.token
+        }).then(scmInfo =>
+            new Promise((resolve, reject) => {
+                this.breaker.runCommand({
+                    action: 'getCommit',
+                    token: config.token,
+                    params: {
+                        owner: scmInfo.user,
+                        repo: scmInfo.repo,
+                        sha: config.sha
+                    }
+                }, (error, data) => {
+                    if (error) {
+                        return reject(error);
+                    }
+
+                    const commitInfo = data.commit;
+
+                    return resolve({
+                        author: {
+                            avatar: data.author.avatar_url,
+                            name: commitInfo.author.name,
+                            url: data.author.html_url,
+                            username: data.author.login
+                        },
+                        message: commitInfo.message,
+                        url: `https://github.com/${scmInfo.user}/${scmInfo.repo}/tree/${config.sha}`
```

Comment:
I think `github.com` should be replaced by `scmInfo.host`; since it could be something like `git.corp.something.com`. You have the host from scmInfo so might as well use that.